### DISCUSSION TITLE
Error in stackSentinel

### DISCUSSION THREAD

**Discussion by riyasmjgeo** [isce-framework/isce2#521] [Q&A] [answered]
Hello All,

Greetings!
I got an error while running '_run_07_pairs_misreg_' in the stackSentinel process-chain. 
`Type:  <class 'isceobj.Sensor.TOPS.TOPSSwathSLCProduct.TOPSSwathSLCProduct'>
<isceobj.Sensor.TOPS.TOPSSwathSLCProduct.TOPSSwathSLCProduct object at 0x7ffb1ed9d0a0>
Running: overlap_withDEM
['--interferogram', '/home/luis/ops/coarse_interferograms/20211212_20211218', '--reference_dir', '/home/luis/ops/coreg_secondarys/20211212', '--secondary_dir', '/home/luis/ops/coreg_secondarys/20211218', '--overlap_dir', '/home/luis/ops/ESD/20211212_20211218']
module overlap_withDEM not found
Traceback (most recent call last):
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/SentinelWrapper.py", line 251, in <module>
    main(args.start,args.end)
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/SentinelWrapper.py", line 242, in main
    cfgParser.runCmd()
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/SentinelWrapper.py", line 54, in runCmd
    func_modules.main(self.funcParams[section])
AttributeError: 'NoneType' object has no attribute 'main'`

The configuration used for stackSentinel is 
`
stackSentinel.py -s ../slc/ -o ../orbit/ -a ~/Work/Sentinel-1_aux/ -w ../ops/ -d ../dem/demLat_S76_S72_Lon_E062_E073.dem.wgs84 -c 3 -z 6 -r 30 -f 0.70 -p hh -b "-74.579018 -73.772415 65 71" -W offset`

In my first try, I tried to get maximum coverage in the bounding box (covered in all the SLC), with parameter _-W interferogram_. Then I shrunk the AOI from all sides to check if it was because of any overlap issue. Then I tried to run the same with _-W offset_ as well. I do get the same error in all the cases and the process doesn't go forward as I am getting another error while running  '_run_08_timeseries_misreg_' as below: 

`Traceback (most recent call last):
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/invertMisreg.py", line 158, in <module>
    main()
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/invertMisreg.py", line 114, in main
    tbase,dateList,dateDict = date_list(overlapPairs)
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/invertMisreg.py", line 50, in date_list
    d1 = datetime.datetime(*time.strptime(dateList[0],"%Y%m%d")[0:5])
IndexError: list index out of range
Traceback (most recent call last):
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/invertMisreg.py", line 158, in <module>
    main()
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/invertMisreg.py", line 114, in main
    tbase,dateList,dateDict = date_list(overlapPairs)
  File "/home/luis/apps/miniconda3/envs/stack/share/isce2/topsStack/invertMisreg.py", line 50, in date_list
    d1 = datetime.datetime(*time.strptime(dateList[0],"%Y%m%d")[0:5])
IndexError: list index out of range
`

I installed ISCE2 using conda and after setting up the PATH in the usual way, the stackSentinel.py application was not included in the path. I had to find the application (stackSentinel.py) path and had to assign that explicitly in the PATH. I am stating this only because I wonder if the error happened because the application was unable to find any *.py file? 
A detailed error report is also attached for reference.
[error.txt](https://github.com/isce-framework/isce2/files/9060559/error.txt)

I can see that the directories ...../misreg/azimuth/dates and ...../misreg/range/dates are empty, so I guess the error occurred in all the pairs. 
Could you please suggest where to start my debugging? 
ISCE VERSION = 2.5.3. I supplied a combination of S1A and S1B in this process.

Thank you

**Comment by BrieCorsa** [comment]
Hey, I'm just looking over your question and it looks like maybe the bounding box argument is out of order in your stackSentinel.py script? You currently list it as 
-b "West East South North"
The lat/lon should be listed as follows: 
-b "South North West East"

Therefore the boundary box is currently not properly lined up with your frame. Maybe see if that works?

**Reply by riyasmjgeo** [reply]
> Hey, I'm just looking over your question and it looks like maybe the bounding box argument is out of order in your stackSentinel.py script? You currently list it as -b "West East South North" The lat/lon should be listed as follows: -b "South North West East"
> 
> Therefore the boundary box is currently not properly lined up with your frame. Maybe see if that works?

I am looking at a location in Antarctica and for that, the bbox coordinates are in correct order. Before creating the run files in stackSentinel.py, it checks for the overlap between the bbox and SAR data. In that check also, all the datasets were shown. So, I strongly believe it is not an issue with the bbox. 

Another thing I have noticed is that I don't have any `***/reference/overlap/IW*/range_bot_**_**.off` files present in my directory. Can that be the root cause? 
```
No range offsets provided
rdict:  {'rangeOff1': '/home/***/ops/reference/overlap/IW1/range_bot_01_02.off', 'rangeOff2': '/home/**/ops/coreg_secondarys/20200121/overlap/IW1/range_bot_01_02.off', 'azimuthOff': '/home/*******/ops/coreg_secondarys/20200121/overlap/IW1/azimuth_bot_01_02.off'}

```

**Comment by comma-never-coma** [comment, marked as answer]
Hey riyasmjgeo, I met the same problem `module overlap_withDEM not found` while running 'run_07_pairs_misreg' and I just fixed it.

- First, I found the PATH '<your installation path>/isce2/contrib/stack/topsStack/overlap_withDEM.py'. Then I made it executable according to `chmod +x overlap_withDEM.py`, and tried to run it `overlap_withDEM.py -h`. It returned an ERROR just like `<home path>/anaconda3/envs/isce2_env/lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found]...`. 
- Then I checked the dynamic library using `strings <home path>/anaconda3/envs/isce2_env/lib/libstdc++.so.6 | grep GLIBC`. Clearly, the results displayed info from GLIBCXX_3.4.1 to GLIBCXX_3.4.29 except GLIBCXX_3.4.30 due to there was only 'libstdc++.so.6.0.29' existing in the 'anaconda/lib' path.
- Finally, running `conda install -c conda-forge libstdcxx-ng` to update it and 'libstdc++.so.6.0.30' is available. Rerunning 'run_07_pairs_misreg' with no problem.

This worked for me, hope it can help u!

**Reply by riyasmjgeo** [reply]
You are a savior @464658329-cmd it worked perfectly. Thank you so much

**Reply by comma-never-coma** [reply]
You are welcome bro!
